### DISCUSSION TITLE
Zed is continuously writing to disk even while idling 🙁

### DISCUSSION THREAD

**Discussion by rohitshriwas** [zed-industries/zed#19486] [Feature Requests]
The resource monitor reports Zed is continuously writing to disk even when I'm not really doing anything in it:

![image](https://github.com/user-attachments/assets/74bb71b5-159a-42d8-aff8-bd6bf09a372f)

I know it's not a ridiculously high writing rate but I'm concerned for my SSD nevertheless. Perhaps there is some setting to make it give up writing when it's not actively saving files? Or it's writing to some pseudo file handle (socket?) which is incorrectly being picked up as disk writes by the resource monitor?

Anyone know anything?

**Comment by scenery** [comment]
I have the same question. The latest version (0.165.4) wrote 1.48GB to the disk after being idle for about 12 hours.

**Comment by stonebyte** [comment]
According to `fatrace` , zed contiously writes to `~/.local/share/zed/db/0-stable/db.sqlite-wal`

**Reply by rohitshriwas** [reply]
That's a great find. I was able to mount a tmpfs filesystem on that path and that finally got the continuous write off my disk. I know Zed will forget what projects or files were open on reboot due to this but that's an acceptable trade off.

Thank you.

**Comment by SomeoneToIgnore** [comment]
This comment have found a culprit in the code: https://github.com/zed-industries/zed/issues/16472#issuecomment-2569210289

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
